Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/LocationUpdateService.java

```diff
@@ -368,8 +374,159 @@ private void updateDriverStatus(Boolean status) {
+                JSONObject pt = waypointArray.getJSONObject(i);
+                latLngPoints.add(new LatLng((Double) pt.get("lat"), (Double) pt.get("lon")));
+            }
+
+            int resultIndex = PolyUtil.locationIndexOnEdgeOrPath(new LatLng(latitude, longitude), latLngPoints, PolyUtil.isClosedPolygon(latLngPoints), true, 30.0);
+
+            if (resultIndex == -1) {
+
+                String devCount = getValueFromStorage("RIDE_WAYPOINT_DEVIATION_COUNT");
+
+                if(devCount == null){
+                    return;
+                }
+
+                int deviation = Integer.parseInt(devCount);
+                deviation = deviation + 1;
+                updateStorage("RIDE_WAYPOINT_DEVIATION_COUNT", String.valueOf(deviation));
+
+                Log.d("new deviation count" , String.valueOf(deviation));
+            } else {
+               Log.d(  "in path", String.valueOf(resultIndex));
+            }
+        }
+        catch (Exception e) {
+            Log.e("deviation update exception", e.toString());
+        }
+    }
+
     /*Location update API call*/
-    private void callDriverCurrentLocationAPI(double latitude, double longitude, float accuracy, String locTime, String log, String locationSource, String triggerFunctionValue) {
+    private void callDriverCurrentLocationAPI(double latitude, double longitude, float accuracy, String locTime, String log, String locationSource, String triggerFunctionValue)  {
+        try {
+            String localStage = getValueFromStorage("LOCAL_STAGE");
+
+            if (localStage.equals("RideStarted")) {
+                if (rideWaypoints == null) {
+                    ExecutorService executor1 = Executors.newSingleThreadExecutor();
+                    executor1.execute(() -> {
+                        try {
+                            Handler routeHandler = new Handler(Looper.getMainLooper());
+                            String rideStartLat = getValueFromStorage("RIDE_START_LAT");
+                            String rideStartLon = getValueFromStorage("RIDE_START_LON");
+                            String rideEndLat = getValueFromStorage("RIDE_END_LAT");
+                            String rideEndLon = getValueFromStorage("RIDE_END_LON");
+
+                            if(rideStartLat == null || rideStartLon == null || rideEndLat == null || rideEndLon == null) {
+                                return;
+                            }
+
+                            System.out.println(Double.parseDouble(rideStartLat) + " " + Double.parseDouble(rideStartLon) + " " + Double.parseDouble(rideEndLat) + " " + Double.parseDouble(rideEndLon) + "fetched vals");
+
+                            getRoute(Double.parseDouble(rideStartLat), Double.parseDouble(rideStartLon), Double.parseDouble(rideEndLat), Double.parseDouble(rideEndLon));
+                            routeHandler.post(() -> {
+                                try {
```

Comment:
No need to have separate try/catch block on this function, just remove throws exception from the function